delfin/drivers/hitachi/hnas/nas_handler.py

```diff
@@ -0,0 +1,825 @@
+                        'location': None,
+                        'soft_version': None,
+                        'cpu_info': None,
+                        'memory_size': None,
+                        'mgmt_ip': None
+                    }
+                    controller_list.append(controller_model)
+            return controller_list
+        except exception.DelfinException as e:
+            err_msg = "Failed to get controllers from " \
+                      "hitachi nas: %s" % (six.text_type(e.msg))
+            LOG.error(err_msg)
+            raise e
+        except Exception as err:
+            err_msg = "Failed to get controllers from " \
+                      "hitachi nas: %s" % (six.text_type(err))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+
+    def list_alerts(self, query_para):
+        try:
+            alert_info = self.ssh_do_exec([constant.ALERT_INFO_COMMAND])
+            alert_array = self.get_table_data(alert_info)
+            alert_list = []
+            for alert in alert_array:
+                value_array = alert.split()
+                if len(value_array) > 4 \
+                        and '******' not in value_array[0] \
+                        and value_array[1] in constant.SEVERITY_MAP:
+                    occur_time = \
+                        value_array[2] + ' ' + \
+                        value_array[3].split("+")[0]
+                    occur_time = \
+                        int(time.mktime(time.strptime(
+                            occur_time, constant.TIME_TYPE))) * 1000
+                    if not query_para or \
+                            (int(query_para['begin_time'])
+                             <= occur_time
+                             <= int(query_para['end_time'])):
+                        description = ''
+                        for i in range(4, len(value_array)):
+                            description += value_array[i] + ' '
+                        severity = constant.SEVERITY_MAP.get(value_array[1])
+                        alert_model = {
+                            'alert_id': value_array[0],
+                            'alert_name': value_array[0],
+                            'severity': severity,
+                            'category': constants.Category.FAULT,
+                            'type': constants.EventType.EQUIPMENT_ALARM,
+                            'occur_time': occur_time,
+                            'description': description,
+                            'match_key': hashlib.md5(
+                                (value_array[0] + severity +
+                                 description).encode()).hexdigest(),
+                            'resource_type': constants.DEFAULT_RESOURCE_TYPE,
+                            'location': ''
+                        }
+                        alert_list.append(alert_model)
+            return alert_list
+        except exception.DelfinException as e:
```

Comment:
Almost every function has these two exceptions, and the difference between these exceptions is the content of `err_msg`, so we could consider extracting out it.